File: Python/169.py
class Solution:
    # @param {integer[]} nums
    # @return {integer}
    """ Version 1 First Try
    def majorityElement(self, nums):
        n = len(nums)/2
        dic = {}
        for i in nums:
            if i not in dic:
                dic[i] = 0
            dic[i] += 1
            if dic[i] > n:
                return i
        return 0
    """
    # Version 2 Concise
    def majorityElement(self, nums):
        return sorted(nums)[len(nums) / 2]
    

File: C++/169.cpp
class Solution {
public:
    /* Concise Solution
    int majorityElement(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        return nums[nums.size()/2];
    }
    */
    
    // Vote Solution
    int majorityElement(vector<int>& nums) {
        int majority = 0; // count = 0, it does not matter;
        int count = 0;
        for(int num : nums){
            if(majority == num) count++;
            else if(count == 0){
                majority = num;
                count = 1;
            }
            else count--;
        }
        return majority;
    }    
};